### PR TITLE
feat(notification): add simple SmartThings switch activation

### PR DESCRIPTION
**Pull request by pmilano1**
### Description

Added basic switch activation for SmartThings, will turn on switch based on the below env vars. Motivation being I work in the dark, and wanted to turn a light on as a notification.

SMARTTHINGS_TOKEN=
SMARTTHINGS_SWITCH_LABEL=

### Testing

- Tested with each of the above vars empty - no activation attempt
- Tested with each of the above incorrect - warning messages about incorrect var
- Tested with each of the above correct - on come the lights!

### New dependencies

-     "@bridgerakol/samsung-smart-api": "^2.8.1"

### CODE REVIEW

**Comment by jef** [comment]
> Added so that linting works properly on windows.

I've been wanting to know how to fix this! Thank you.

**Comment by pmilano1** [comment]
> > Added so that linting works properly on windows.
> 
> I've been wanting to know how to fix this! Thank you.

Ideally I would have done it prior to my horrible commit chain - but I felt challenged by it :)

File: src/notification/smartthings.ts [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+          if (smartthings.device === item.label) {
+            match = true;
+            let device_status = (await st.devices.getStatus(item.deviceId)).data
+              .components.main.switch.switch.value;
+            if (device_status === 'on') {
+              device_status = true;
+            } else device_status = false;
+            if (!device_status) {
+              logger.debug(`Turning on ${smartthings.device}`);
+              st.devices.commands(item.deviceId, 'on');
```
Comment (jef) [inline src/notification/smartthings.ts:32]:
```suggestion
            if (device_status !== 'on') {
              logger.debug(`Turning on ${smartthings.device}`);
              st.devices.commands(item.deviceId, 'on');
```

Can we reduce this to this?

**Review by jef — COMMENTED** [review]
Overall looks great! Thanks for the implementation. Just left a few comments if you don't mind addressing :relaxed:

File: src/notification/smartthings.ts [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+export async function activateSmartthingsSwitch() {
+  if (
+    ('token' in smartthings &&
+      !smartthings.token &&
+      'device' in smartthings &&
+      !smartthings.device) ||
+    (!('token' in smartthings) && !('device' in smartthings))
+  ) {
+    return;
+  }
```
[the remark this answers is not in the document]
Comment (pmilano1) [reply src/notification/smartthings.ts:16]: I think I should have actually written in something incase smartthings wasn't present at all - maybe you've accounted for this already. I'm good if you're good man.

[the remark this answers is not in the document]
Comment (jef) [reply src/notification/smartthings.ts]: I believe `smartthings` is defined, but with undefined properties. So I believe this will work!

**Review by jef — APPROVED** [review]
Awesome! Lets pull this in. Thanks for everything!

**Comment by jef** [comment]
YAY!!!

**Pull request merged**